instrumentation-tests/src/androidTest/java/com/mapbox/navigation/instrumentation_tests/core/CopilotIntegrationTest.kt

```diff
@@ -0,0 +1,148 @@
+import com.mapbox.navigation.core.directions.session.RoutesExtra
+import com.mapbox.navigation.core.internal.HistoryRecordingStateChangeObserver
+import com.mapbox.navigation.core.internal.extensions.registerHistoryRecordingStateChangeObserver
+import com.mapbox.navigation.core.trip.session.NavigationSessionState
+import com.mapbox.navigation.instrumentation_tests.activity.EmptyTestActivity
+import com.mapbox.navigation.instrumentation_tests.utils.MapboxNavigationRule
+import com.mapbox.navigation.instrumentation_tests.utils.coroutines.clearNavigationRoutesAndWaitForUpdate
+import com.mapbox.navigation.instrumentation_tests.utils.coroutines.routesUpdates
+import com.mapbox.navigation.instrumentation_tests.utils.coroutines.sdkTest
+import com.mapbox.navigation.instrumentation_tests.utils.coroutines.setNavigationRoutesAndAwaitError
+import com.mapbox.navigation.instrumentation_tests.utils.history.MapboxHistoryTestRule
+import com.mapbox.navigation.instrumentation_tests.utils.routes.MockRoute
+import com.mapbox.navigation.instrumentation_tests.utils.routes.RoutesProvider
+import com.mapbox.navigation.testing.ui.BaseTest
+import com.mapbox.navigation.testing.ui.utils.getMapboxAccessTokenFromResources
+import com.mapbox.navigation.testing.ui.utils.runOnMainSync
+import com.mapbox.navigation.utils.internal.logE
+import kotlinx.coroutines.flow.filter
+import kotlinx.coroutines.flow.first
+import org.junit.Assert.assertEquals
+import org.junit.Before
+import org.junit.Rule
+import org.junit.Test
+
+@OptIn(ExperimentalPreviewMapboxNavigationAPI::class)
+class CopilotIntegrationTest : BaseTest<EmptyTestActivity>(EmptyTestActivity::class.java) {
+
+    private lateinit var mapboxNavigation: MapboxNavigation
+    private lateinit var mockRoute: MockRoute
+
+    @get:Rule
+    val mapboxNavigationRule = MapboxNavigationRule()
+    @get:Rule
+    val mapboxHistoryTestRule = MapboxHistoryTestRule()
+
+    override fun setupMockLocation(): Location {
+        mockRoute = RoutesProvider.dc_very_short(activity)
+        return mockLocationUpdatesRule.generateLocationUpdate {
+            latitude = mockRoute.routeWaypoints.first().latitude()
+            longitude = mockRoute.routeWaypoints.first().longitude()
+        }
+    }
+
+    @Before
+    fun setUp() {
+        runOnMainSync {
+            mapboxNavigation = MapboxNavigationProvider.create(
+                NavigationOptions.Builder(activity)
+                    .accessToken(getMapboxAccessTokenFromResources(activity))
+                    .build()
+            )
+            mapboxHistoryTestRule.historyRecorder = mapboxNavigation.historyRecorder
+        }
+    }
+
+    @Test
+    fun copilotObserverReceivesSameIdsAsDeveloperMetadata() = sdkTest {
+        val collectedDeveloperMetadatas = mutableListOf<DeveloperMetadata>()
+        val developerMetadataObserver = DeveloperMetadataObserver {
+            Log.i("[DeveloperMetadataObserver]", "collected metadata: $it")
```

Comment:
Curious, why are we adding logs to a test? Are these useful?

Reply:
It was useful when I was using flows and the error looked like "Test timed out" instead of "expected 7 elements, found 6". Removed.